feat(scheduler): Frequent job scheduling

adityahase opened the pull request:
The minimum configurable frequency for scheduled jobs, as of now is 60 seconds, as defined by 

```python3
schedule.every(60).seconds.do(enqueue_events_for_all_sites)
```
This PR attempts to circumvent this limit and allow sub-minute frequencies without affecting existing configurations by making it configurable with `scheduler_tick_interval` key in `common_site_config.json`

e.g. Now a job can be scheduled once every 15 seconds with 
```python3
scheduler_events = {
	"cron": {
		"* * * * * */15": [ ... ],
	}
}

```
 in `hooks.py` and
```JSON
"scheduler_tick_interval": 15
```
 or lower in `common_site_config.json`

Documentation PR: https://github.com/frappe/frappe_io/pull/261

surajshetty3416 commented on frappe/utils/scheduler.py:
```diff
@@ -24,7 +24,7 @@ def start_scheduler():
 	Specify scheduler_interval in seconds in common_site_config.json'''
 
-	schedule.every(60).seconds.do(enqueue_events_for_all_sites)
+	schedule.every(frappe.get_conf().scheduler_tick_interval or 60).seconds.do(enqueue_events_for_all_sites)
```
This cannot be greater than `fast_job_interval` or else `fast_job_interval` will be pointless.

For fresh installation where `fast_job_interval` & `scheduler_tick_interval` are not set, each "Fast" job will be executed after 1min (but the default interval for "Fast" job is 30secs)

Consider adding a check for this case.

adityahase replied:
Removed `Fast` job types and `fast_job_interval` these can now be used with `cron`

e.g. `* * * * * */5` to schedule a job every 5 seconds. Of course, for this to work as expected `scheduler_tick_interval` must be set appropriately.

adityahase committed:
feat(scheduler): Frequent job scheduling

Scheduler tick interval (default 60 seconds) can now be configured with scheduler_tick_interval key in common_site_config.json
